src/transformers/utils/model_parallel_utils.py

```diff
@@ -12,10 +12,176 @@
+                for name in elem.name:
+                    copy_elem = copy.deepcopy(elem)
+                    copy_elem.name = name
+                    cache_tp_mapping[cls].append(copy_elem)
+
+        self.__MAPPING__ = {cls: {} for cls in cache_tp_mapping}
+        # clear exist mapping rather than making new mapping dict
+
+        for cls, mapping in cache_tp_mapping.items():
+            for elem in mapping:
+                if elem.code in self.__MAPPING__[cls]:
+                    self.__MAPPING__[cls][elem.code].append(elem)
+                else:
+                    self.__MAPPING__[cls][elem.code] = [elem]
+
+    @staticmethod
+    def _load_class_by_model_name(model_name):
+        transformers = importlib.import_module("transformers")
+        cls = getattr(transformers, f"{model_name}PreTrainedModel", None)
+        if cls is None:
+            cls = getattr(transformers, f"{model_name}PretrainedModel", None)
+        assert cls is not None, f"Can not import the model named {cls}."
+        return cls
+
+    def get_mapping(self, model):
+        for cls, mapping in self.__MAPPING__.items():
+            if isinstance(model, cls):
+                return mapping
+        return None
+
+    def column_parallel_params(self, model):
+        mapping = self.get_mapping(model)
+        if mapping is not None:
+            return mapping["Col"]
+
+    def row_parallel_params(self, model):
+        mapping = self.get_mapping(model)
+        if mapping is not None:
+            return mapping["Row"]
+
+    def update_attrs(self, model):
+        mapping = self.get_mapping(model)
+        if mapping is not None:
+            return mapping["Update"]
+
+    def search(self, model, param_name):
+        mapping = self.get_mapping(model)
+        count_contain_elem_in_param = 0
+        param_split = param_name.split(".")
+        first_check = []
+
+        for code, elems in mapping.items():
+            for elem in elems:
+                if elem.name in param_name:
+                    first_check.append(elem)
+
+        for elem in first_check:
+            elem_split = elem.name.split(".")
+            for split in elem_split:
+                if split in param_split:
+                    count_contain_elem_in_param += 1
+            if count_contain_elem_in_param == len(elem_split):
+                return elem
+
+        return None
+
+    def is_combined_qkv_param(self, model, param_name):
+        elem = self.search(model, param_name)
+        if elem is not None:
+            return elem.combined_qkv
+
+    def get_combined_qkv_degree(self, model, param_name, module):
+        if self.is_combined_qkv_param(model, param_name) and hasattr(module, "weight"):
+            bigger = max(module.weight.size(0), module.weight.size(1))
+            smaller = min(module.weight.size(0), module.weight.size(1))
+            return bigger // smaller
+        return 1
+
+    def is_reversed_param(self, model, param_name):
```

Comment:
Nit: For the `is_X` functions, I think it would be helpful to type annotate it with `def is_X() -> bool:` and add `return False` by default.

Reply:
Originally, all methods have docstring, but some errors related to documentation blocked the CI. Could you help me?